lib/util/addExtraParamsTo.js

```diff
@@ -10,7 +10,11 @@ function addExtraParamsTo(routeArguments) {
     //  is null. WWARNER 08/21/2014
     paramsAndRequest.pop();
 
-    var extraParams = Array.prototype.slice.call(arguments, 1);
```

Comment:
tried it with setting the size up front. i think it's uglier and harder to follow what's going on:

```js
function addExtraParamsTo(routeArguments) {
    var paramsAndRequest = routeArguments.slice(0);

    // The last parameter in routeArguments is always null because
    //  of Backbone request parsing. So here we just replace that
    //  last parameter with the request. Developers don't usually
    //  notice that the last parameter passed to their route handler
    //  is null. WWARNER 08/21/2014
    paramsAndRequest.pop();

    var len = arguments.length;
    var extraParams = new Array(Math.max(0, len - 1));

    for (var i = 1; i < len; i++) {
        extraParams[i - 1] = arguments[i];
    }

    return paramsAndRequest.concat(extraParams);
}
```

what do you think?